src/ResourceManager/Insights/Commands.Insights/Diagnostics/SetAzureRmDiagnosticSettingCommand.cs

```diff
@@ -57,9 +78,9 @@ public class SetAzureRmDiagnosticSettingCommand : ManagementCmdletBase
         /// <summary>
         /// Gets or sets the enable parameter of the cmdlet
         /// </summary>
-        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The value indicating whether the diagnostics should be enabled or disabled")]
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The value indicating whether the diagnostics should be enabled or disabled")]
         [ValidateNotNullOrEmpty]
-        public bool Enabled { get; set; }
+        public bool? Enabled { get; set; }
```

Comment:
Customer can run the following
Set-AzureRmDiagnosticSettings -ResourceId <> -DisableStorage